Resources/config/services.yml

```diff
@@ -9,32 +12,56 @@ parameters:
     support_tools.info_collectors.symfony.kernel.config.class: EzSystems\EzSupportToolsBundle\SystemInfo\Collector\ConfigurationSymfonyKernelSystemInfoCollector
 
 services:
+    support_tools.command.dump_info:
+        class: %support_tools.command.dump_info.class%
+        arguments:
+            - @support_tools.info_collector.registry
+        tags:
+            - { name: console.command }
+
+    support_tools.info_collector.registry:
+        class: %support_tools.info_collector.registry.class%
+
     support_tools.info_collector.system_info.ezc:
         class: ezcSystemInfo
-        factory: [EzSystems\EzSupportToolsBundle\SystemInfo\EzcSystemInfoFactory, buildEzcSystemInfo]
+        factory: [@support_tools.info_collector.system_info.ezc.factory, buildEzcSystemInfo]
+
+    support_tools.info_collector.system_info.ezc.factory:
+        class: %support_tools.info_collector.system_info.ezc.factory.class%
+        lazy: true
 
     support_tools.info_collectors.composer.lock_file:
         class: %support_tools.info_collectors.composer.lock_file.class%
         arguments:
             - "%kernel.root_dir%/../composer.lock"
+        tags:
+            - { name: "support_tools.info_collector", alias: "composer" }
 
     support_tools.info_collectors.database.doctrine:
         class: %support_tools.info_collectors.database.doctrine.class%
         arguments:
             - @database_connection
+        tags:
+            - { name: "support_tools.info_collector", alias: "database" }
 
     support_tools.info_collectors.hardware.ezc:
         class: %support_tools.info_collectors.hardware.ezc.class%
         arguments:
             - @support_tools.info_collector.system_info.ezc
+        tags:
+            - { name: "support_tools.info_collector", alias: "hardware" }
 
     support_tools.info_collectors.php.ezc:
         class: %support_tools.info_collectors.php.ezc.class%
         arguments:
             - @support_tools.info_collector.system_info.ezc
+        tags:
+            - { name: "support_tools.info_collector", alias: "php" }
```

Comment:
We also need to rename the service tag, btw. `support_tools.system_info_collector` ?

Also, we use `identifier` in the registry's code, not `alias`.